mocks/client.go

```diff
@@ -0,0 +1,2734 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: client/client.go
+
+// Package mock_client is a generated GoMock package.
+package mock_client
```

Comment:
I know we agreed on having the mocks here, but on seeing them I'm having second thoughts. It may be better for us to move them to the SDK?

Will adding them there require any extra dependencies?

Reply:
Ideally, mockgen can allow us to specify external package (in this case the SDK) and we can generate the mock package as part of make test (and not push it the generated code in).  However, we are not able to do that :(